### ISSUE TITLE
[BUG] Validation split --valid-split "validation" not working

### ISSUE THREAD

**Issue by Stealthwriter** [huggingface/autotrain-advanced#462]
### Prerequisites

- [X] I have read the [documentation](https://hf.co/docs/autotrain).
- [X] I have checked other issues for similar problems.

### Backend

Local

### Interface Used

CLI

### CLI Command

autotrain llm \
  --text_column "text" \
  --warmup_ratio 0.1 \
  --optimizer "adamw_torch" \
  --scheduler "linear" \
  --weight_decay 0 \
  --block_size 256 \
  --peft \
  --lora_r 8 \
  --lora_alpha 16 \
  --lora_dropout 0.05 \
  --logging_steps -1 \
  --evaluation_strategy "epoch" \
  --save_total_limit 1 \
  --save_strategy "epoch" \
  --auto_find_batch_size \
  --mixed-precision "fp16" \
  --quantization "int4" \
  --model_max_length 256 \
  --target_modules "q_proj, k_proj, v_proj, o_proj, gate_proj, up_proj, down_proj, lm_head" \
  --merge_adapter \
  --padding "right" \
  --train \
  --token  hidden\
  --repo-id hidden
  --push-to-hub \
  --model "mistralai/Mistral-7B-Instruct-v0.1" \
  --project-name "malakZ" \
  --epochs 3 \
  --gradient-accumulation 4 \
  --disable_gradient_checkpointing \
  --lr 0.00003 \
  --log "wandb" \
  --data-path "./data" \
  --train-split "train" \
  --valid-split "valid" \
  --batch-size 2 \
  --max-grad-norm 1

### Error Logs

tralai/Mistral-7B-Instruct-v0.1', project_name='malakZ2', seed=42, epochs=8, gradient_accumulation=4, disable_gradient_checkpointing=True, lr=3e-05, log='wandb', data_path='./data', train_split='train', valid_split='valid', batch_size=2, func=<function run_llm_command_factory at 0x7ff428f82830>)
> INFO    Dataset: malakZ2 (lm_training)
Train data: ['./data/train.csv']
Valid data: ['./data/valid.csv']
Column mapping: {'text': 'text', 'rejected_text': 'rejected', 'prompt': 'prompt'}

Saving the dataset (1/1 shards): 100%|█████████████████████████████████████████████████████████████████████████████████████████████████| 8510/8510 [00:00<00:00, 2157099.60 examples/s]
Saving the dataset (1/1 shards): 100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 448/448 [00:00<00:00, 270055.79 examples/s]
> INFO    Starting local training...
> INFO    {"model":"mistralai/Mistral-7B-Instruct-v0.1","project_name":"malakZ2","data_path":"malakZ2/autotrain-data","train_split":"train","valid_split":null,"add_eos_token":false,"block_size":256,"model_max_length":256,"padding":"right","trainer":"default","use_flash_attention_2":false,"log":"wandb","disable_gradient_checkpointing":true,"logging_steps":-1,"evaluation_strategy":"epoch","save_total_limit":1,"save_strategy":"epoch","auto_find_batch_size":true,"mixed_precision":"fp16","lr":0.00003,"epochs":8,"batch_size":2,"warmup_ratio":0.1,"gradient_accumulation":4,"optimizer":"adamw_torch","scheduler":"linear","weight_decay":0.0,"max_grad_norm":1.0,"seed":42,"apply_chat_template":false,"quantization":"int4","target_modules":"q_proj, k_proj, v_proj, o_proj, gate_proj, up_proj, down_proj, lm_head","merge_adapter":true,"peft":true,"lora_r":8,"lora_alpha":16,"lora_dropout":0.05,"model_ref":null,"dpo_beta":0.1,"prompt_text_column":"autotrain_prompt","text_column":"autotrain_text","rejected_text_column":"autotrain_rejected_text","push_to_hub":true,"repo_id":"hidden","username":null,"token":"hidden"}
> INFO    ['accelerate', 'launch', '--num_machines', '1', '--num_processes', '1', '--mixed_precision', 'fp16', '-m', 'autotrain.trainers.clm', '--training_config', 'malakZ2/training_params.json']
The following values were not passed to `accelerate launch` and had defaults used instead:
        `--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
/usr/local/lib/python3.10/dist-packages/trl/trainer/ppo_config.py:141: UserWarning: The `optimize_cuda_cache` arguement will be deprecated soon, please use `optimize_device_cache` instead.
  warnings.warn(
🚀 INFO   | 2024-01-13 19:35:41 | __main__:process_input_data:41 - loading dataset from disk
🚀 INFO   | 2024-01-13 19:35:41 | __main__:process_input_data:82 - Train data: Dataset({
    features: ['autotrain_text'],
    num_rows: 8510
})
🚀 INFO   | 2024-01-13 19:35:41 | __main__:process_input_data:83 - Valid data: None
Loading checkpoint shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:08<00:00,  4.33s/it]
🚀 INFO   | 2024-01-13 19:35:53 | __main__:train:277 - Using block size 256
Running tokenizer on train dataset: 100%|████████████████████████████████████████████████████████████████████████████████████████████████| 8510/8510 [00:00<00:00, 23463.50 examples/s]
Grouping texts in chunks of 256 (num_proc=4): 100%|██████████████████████████████████████████

### Additional Information

Why valid data is none, I assigned it though

**Comment by jackshiwl** [comment]
I have the same issue as you.

**Comment by abhishekkrthakur** [comment]
validation data is not used for llm training as validation is not representative of model's performance. you can validate manually after training

**Comment by jackshiwl** [comment]
thanks for the swift response. 

does that mean, it is redundant to pass a `valid.csv` file into autotrain? 

Simply fine-tune on train.csv, and manually validate after training like you said?

**Comment by abhishekkrthakur** [comment]
yes

**Comment by Stealthwriter** [comment]
That would take alot of time, Open AI shows you the validation directly
while fine tuning which gives you an idea if the training is going good or
not.

**Comment by abhishekkrthakur** [comment]
we can enable validation but it wont be representative. we also have to think about all the models and params combinations. openai doesnt have to do that. ill come back to this in a couple of days after some internal experiments.

**Comment by yihahn** [comment]
Hi,  @abhishekkrthakur  
I guess that the validation process could be effective when it comes to fine-tuning LLM with the task of generating short answers consisting of less than 5 words. Is there any plan to bring back validation argument in the middle of LLM fine-tuning?
